src/kixi/datastore/web_server.clj

```diff
@@ -40,6 +40,31 @@
                               (concat [i] interceptors)
                               [i])))))
 
+(spec/def ::context
+  (spec/keys :req []
+             :opts []))
+
+(spec/def ::error #{:schema-invalid-name :schema-invalid-definition :unknown-schema})
+(spec/def ::msg (spec/keys :req []
+                           :opts []))
+(spec/def ::error-map
+  (spec/keys :req [::error ::msg]))
+
+(spec/fdef return-error
+        :args (spec/cat :ctx ::context
+                        :args (spec/alt :error-map ::error-map
+                                        :error-parts (spec/cat :error ::error
+                                                               :msg ::msg))))
```

Comment:
well the return value is the whole context, this just sploges something into it. I guess i could add a :ret for that sub-part of it...